static/app/components/dataExport.tsx

```diff
@@ -17,59 +18,47 @@ export enum ExportQueryType {
   Discover = 'Discover',
 }
 
-type DataExportPayload = {
+interface DataExportPayload {
   queryType: ExportQueryType;
   queryInfo: any; // TODO(ts): Formalize different possible payloads
-};
+}
 
-type Props = {
+interface DataExportProps {
   api: Client;
   disabled?: boolean;
   organization: Organization;
   payload: DataExportPayload;
   icon?: React.ReactNode;
-};
-
-type State = {
-  inProgress: boolean;
-};
+  children?: React.ReactNode;
+}
 
-class DataExport extends React.Component<Props, State> {
-  state = this.initialState;
+function DataExport({
+  api,
+  children,
+  disabled,
+  organization,
+  payload,
+  icon,
+}: DataExportProps): React.ReactElement {
+  const previousPayload = usePrevious(payload);
+  const [inProgress, setInProgress] = React.useState(false);
 
-  componentDidUpdate({payload: prevPayload}) {
-    const {payload} = this.props;
-    if (!isEqual(prevPayload, payload)) {
-      this.resetState();
+  React.useEffect(() => {
+    if (!isEqual(payload, previousPayload)) {
```

Comment:
@leeandher I dont know the exact usage to this, but could this effect be written without the isEqual?
```tsx
React.useEffect(() => {
  setInProgress(false)
},[payload.queryType, payload.queryInfo])
```

Reply:
@JonasBa Yep, that looks good! Out with `isEqual` 🔥